src/Analyzer.Utilities/JsonLineNumberResolver.cs

```diff
@@ -55,7 +55,15 @@ public int ResolveLineNumber(string pathInExpandedTemplate)
             // even the first property could not be found in the original template.
             if (tokenFromOriginalTemplate.Equals(originalTemplateRoot))
             {
-                // See if the path starts with indexing into the template resources[] array
+                return 0;
+            }
+
+            // If the JToken returned from looking up the expanded template path is
+            // pointing to the resources array of the original template, then it's likely
+            // the expanded template path used an index too large for the original template.
+            if (tokenFromOriginalTemplate.Equals(originalTemplateRoot.InsensitiveToken("resources")))
+            {
+                // Verify the path starts with indexing into the template resources[] array
```

Comment:
This small code change was needed because of the algorithm change when accounting for wildcards.

Previously (for example), if the first property in the expanded path was *resources[5]* and the original template only had resources 0-3, `originalTemplateRoot.InsensitiveToken()` would fail to find exactly *resources[5]*, and so would return the root node.

Now, `InsensitiveToken()` will find *resources*, but not *element 5*, and will return the node representing the whole *resources* array.